softlayer/vm/softlayer_creator.go

```diff
@@ -68,7 +97,7 @@ func (c SoftLayerCreator) Create(agentID string, stemcell bslcstem.Stemcell, clo
 		return SoftLayerVM{}, bosherr.WrapError(err, "Creating VirtualGuestService from SoftLayer client")
 	}
 
-	virtualGuest, err := virtualGuestService.CreateObject(virtualGuestTemplate)
+	virtualGuest, err := virtualGuestService.CreateObject(*c.checkValid(&virtualGuestTemplate))
 	if err != nil {
```

Comment:
I really don't like this statement. It should be broken into two:

1. check for validity of template and err if not valid.
2. Call to `CreateObject` once valid

Also, the `checkValid` should not be on `*c` and not expect `&virtualGuestTemplate` instead, just check and return error or false as soon as their is an error that you cannot recover from. If you need help on this then please let me know o talk to @zhang-hua